ui/components-react/src/test/tree/controlled/TreeNodeLoader.test.ts

```diff
@@ -278,11 +329,84 @@ describe("PagedTreeNodeLoader", () => {
         const loadedIds = await extractLoadedNodeIds(loadObs);
         expect(loadedIds).to.be.deep.eq(["1", "1-1", "1-2", "2"]);
       });
+    });
+  });
+});
 
+describe("AbstractTreeNodeLoader", () => {
+  describe("loadNode", () => {
+    it("accepts non-rxjs observables", async () => {
+      const modelSource = new TreeModelSource();
+      const promise = new ResolvablePromise<void>();
+      const nodeLoader = createCustomTreeNodeLoader(
+        modelSource,
+        () => {
+          return {
+            [Symbol.observable]() {
+              return this;
+            },
+            subscribe(
+              observerOrNext?: Observer<LoadedNodeHierarchy> | ((value: LoadedNodeHierarchy) => void) | null,
+              _error?: ((error: any) => void) | null,
+              complete?: (() => void) | null,
+            ) {
+              if (typeof observerOrNext === "object") {
+                observerOrNext?.complete?.();
+              } else {
+                complete?.();
+              }
+              return { closed: true, add() { }, unsubscribe() { } };
+            },
+          };
+        }
+      );
+      nodeLoader.loadNode(modelSource.getModel().getRootNode(), 0).subscribe({ complete: () => promise.resolve() });
+      await promise;
     });
 
-  });
+    it("allows one active load at a time", async () => {
+      const modelSource = new TreeModelSource();
+      sinon.stub(modelSource, "modifyModel");
+      const dataProvider = sinon.fake(() => new ResolvablePromise());
+
+      const nodeLoader = createCustomTreeNodeLoader(
+        modelSource,
+        () => defer(() => rxjsFrom<Promise<LoadedNodeHierarchy>>(dataProvider())),
+      );
+      nodeLoader.loadNode(modelSource.getModel().getRootNode(), 0).subscribe();
+      nodeLoader.loadNode(modelSource.getModel().getRootNode(), 1).subscribe();
+
+      await Promise.resolve();
+      expect(dataProvider).to.have.been.calledOnce;
 
+      const hierarchy: LoadedNodeHierarchy = {
+        parentId: undefined,
+        offset: 0,
+        numChildren: undefined,
+        hierarchyItems: [],
+      };
+      await dataProvider.firstCall.returnValue.resolve(hierarchy);
+      expect(dataProvider).to.have.been.calledTwice;
+    });
+
+    function createCustomTreeNodeLoader(
```

Comment:
Does not look like this function uses anything deprecated? Are all these lines to disable deprecation rule needed?

Reply:
I deprecated `AbstractTreeNodeLoader` at one point and forgot to remove these. Thanks for pointing them out, fixed in 68df2a896779ac7f8302f8b28c8d85e1e95c3484.